pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -639,3 +640,80 @@ func buildFallthroughNetworkFilters(push *model.PushContext, node *model.Proxy)
 
 	return filterStack
 }
+
+// TODO: This code is still insufficient. Ideally we should be parsing all the virtual services
+// with TLS blocks and build the appropriate filter chain matches and routes here. And then finally
```

Comment:
>we should be parsing all the virtual services with TLS blocks and build the appropriate filter chain matches and routes here.

We already iterate over all the virtual services while running through `buildSidecarOutboundListeners`. What is being missed in this iteration over virtual services.
https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/tls.go#L125-L152

Reply:
see https://github.com/istio/istio/issues/21170 .. we need to setup a listener on 0.0.0.0:Port to trap all TLS traffic like https://example.com and be able to route it to internal services like productpage.ns1.svc.cluster.local . Today, we setup the listener with TLS routing only on clusterIPofProductpage:Port. We need to setup these routes on the catch all listeners as well. Same goes for HTTP traffic